deploy/webhook.yaml

```diff
@@ -0,0 +1,258 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: service-api
+  labels:
+    name: service-api
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: service-api-admission
+webhooks:
+  - name: validate.networking.x-k8s.io
+    matchPolicy: Equivalent
+    rules:
+      - operations: [ "CREATE" ]
+        apiGroups: [ "networking.x-k8s.io" ]
+        apiVersions: [ "v1alpha1" ]
+        resources: [ "httproutes" ]
+    failurePolicy: Fail
+    sideEffects: None
+    admissionReviewVersions:
+      - v1
+      - v1beta1
+    clientConfig:
+      service:
+        name: service-api-admission
+        namespace: service-api
+        path: "/validate"
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: service-api-admission
+  annotations:
+  labels:
+    name: service-api-webhook
+    version: 0.0.1
+  namespace: service-api
+---
+# Source: ingress-nginx/templates/admission-webhooks/job-patch/clusterrole.yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: service-api-admission
+  labels:
+    name: service-api
+    version: 0.0.1
+rules:
+  - apiGroups:
+      - admissionregistration.k8s.io
+    resources:
+      - validatingwebhookconfigurations
+    verbs:
+      - get
+      - update
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: service-api-admission
+  annotations:
+  labels:
+    name: service-api-webhook
+    version: 0.0.1
+  namespace: service-api
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: service-api-admission
+subjects:
+  - kind: ServiceAccount
+    name: service-api-admission
+    namespace: service-api
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: service-api-admission
+  annotations:
+  labels:
+    name: service-api-webhook
+    version: 0.0.1
+  namespace: service-api
+rules:
+  - apiGroups:
+      - ''
+    resources:
+      - secrets
+    verbs:
+      - get
+      - create
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: service-api-admission
+  annotations:
+  labels:
+    name: service-api-webhook
+    version: 0.0.1
+  namespace: service-api
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: service-api-admission
+subjects:
+  - kind: ServiceAccount
+    name: service-api-admission
+    namespace: service-api
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    name: service-api-webhook
+    version: 0.0.1
+  name: service-api-admission
+  namespace: service-api
+spec:
+  type: ClusterIP
+  clusterIP: 10.96.0.2
```

Comment:
I originally specified this because I was testing some methods of using openssl/cfssl to create self signed certificates, but this probably is not necessary anymore. I will remove it.